File: app/src/main/java/com/kunzisoft/keepass/model/Field.java
package com.kunzisoft.keepass.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Field implements Parcelable {

    private String name;
    private String value;

    public Field(String name, String value) {
        this.name = name;
        this.value = value;
    }

    public Field(Parcel in) {
        this.name = in.readString();
        this.value = in.readString();
    }

    public static final Creator<Field> CREATOR = new Creator<Field>() {
        @Override
        public Field createFromParcel(Parcel in) {
            return new Field(in);
        }

        @Override
        public Field[] newArray(int size) {
            return new Field[size];
        }
    };

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(value);
    }
}

File: app/src/main/java/com/kunzisoft/keepass/model/Field.kt
package com.kunzisoft.keepass.model

import android.os.Parcel
import android.os.Parcelable
import com.kunzisoft.keepass.database.element.security.ProtectedString

class Field : Parcelable {

    var name: String = ""
    var protectedValue: ProtectedString = ProtectedString()

    constructor(name: String, value: ProtectedString = ProtectedString()) {
        this.name = name
        this.protectedValue = value
    }

    constructor(parcel: Parcel) {
        this.name = parcel.readString() ?: name
        this.protectedValue = parcel.readParcelable(ProtectedString::class.java.classLoader) ?: protectedValue
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeParcelable(protectedValue, flags)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Field

        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        return name.hashCode()
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<Field> = object : Parcelable.Creator<Field> {
            override fun createFromParcel(parcel: Parcel): Field {
                return Field(parcel)
            }

            override fun newArray(size: Int): Array<Field?> {
                return arrayOfNulls(size)
            }
        }
    }
}
